strongbox-data-service/src/main/java/org/carlspring/strongbox/data/domain/GenericEntityHook.java

```diff
@@ -36,6 +37,11 @@ public RESULT onRecordBeforeCreate(ORecord iRecord)
         }
 
         ODocument doc = (ODocument) iRecord;
+        if ("ArtifactEntry".equals(doc.getClassName()))
```

Comment:
@mkuligowski we shouldn't change entity data in this hook, because this changes will be not available at client side until explicitly refetch entity from DB. Here you can just validate that `created` field is not empy and throw `OValidationException` in case of invalid field value.
I suggest you to populate `created` field in `ArtifactEntryServiceImpl.save` method.

Reply:
I would also recommend using the class' name, instead of having this as a string like this.

Reply:
@carlspring we have no this class in `strongbox-data-service` module

Reply:
Okay, I see. Thanks for the clarification!